### PR TITLE
feat: add traversable lenses

### PR DESCRIPTION
**Pull request by char0n**
This is POC that traversables can be manipulated via lenses.

This is the [draft of the article](https://www.linkedin.com/pulse/draft/AgGMpHfWPg0aewAAAWLLKtT0R_vAWCIYJMoOR678o56W2vY2_obPLeesnSWdIiPQwQduu30) that I created during the process of creating `RA.lensTraverse`. The last part of `Real World Applications` is still unfinished. Pls feel free to provide review even for the article.

https://www.linkedin.com/pulse/draft/AgGMpHfWPg0aewAAAWLLKtT0R_vAWCIYJMoOR678o56W2vY2_obPLeesnSWdIiPQwQduu30

### CODE REVIEW

File: test/lensTraverse.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    const result = R.over(RA.lensTraverse(RA.Identity.of), R.map(R.add(1)), [
+      RA.Identity.of(2),
+      RA.Identity.of(3),
+    ]);
+    const expected = RA.Identity.of([3, 4]);
+
+    assert.deepEqual(result.value, expected.value);
+  });
+
+  it('should travers left to right', function() {
```
Comment (char0n) [inline test/lensTraverse.js]: traverse

File: src/lensTraverse.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ * @category Relation
+ * @typedef Lens s a = Functor f => (a -> f a) -> s -> f s
+ * @sig Applicative f => (a -> f a) -> Lens s a
+ * @param {!function} of The Applicative-returning function
+ * @return {!function} The Traversable lens
+ * @see {@link http://ramdajs.com/docs/#lens|R.lens}, {@link http://ramdajs.com/docs/#traverse|R.traverse}
+ *
+ * @example
+ *
+ * const maybeLens = RA.lensTraversable(Maybe.of);
```
Comment (srghma) [inline src/lensTraverse.js]: lensTraverse?

Comment (char0n) [reply src/lensTraverse.js]: Yep, it's a typo. Thanks.

**Comment by char0n** [comment]
Article already released here: https://www.linkedin.com/pulse/functional-lenses-javascript-traversables-vladim%C3%ADr-gorej/

**Pull request merged**